vignettes/EnrichmentBrowser.Rmd

```diff
@@ -0,0 +1,990 @@
+
+The `inst/scripts` folder also contains a `README` file that comprehensively documents the usage of both scripts.
+
+# A primer on terminology and statistical theory {#sec:primer}
+
+## Where does it all come from? {#sec:ora}
+
+Test whether known biological functions or processes are
+over-represented (= enriched) in an experimentally-derived gene list,
+e.g. a list of differentially expressed (DE) genes. See [@Goeman07] for
+a critical review.
+
+Example: Transcriptomic study, in which 12,671 genes have been tested
+for differential expression between two sample conditions and 529 genes
+were found DE.
+
+Among the DE genes, 28 are annotated to a specific functional gene set,
+which contains in total 170 genes. This setup corresponds to a
+$2\times2$ contingency table,
+
+```{r deTbl}
+deTable <- matrix(c(28, 142, 501, 12000), 
+                  nrow = 2, 
+                  dimnames = list(c("DE", "Not.DE"), c("In.gene.set", "Not.in.gene.set"))) 
+deTable
+```
+
+where the overlap of 28 genes can be assessed based on the
+hypergeometric distribution. This corresponds to a one-sided version of
+Fisher's exact test, yielding here a significant enrichment.
+
+```{r fisher}
+ fisher.test(deTable, alternative = "greater")
+```
+
+This basic principle is at the foundation of major public and commercial
+enrichment tools such as [*DAVID*](https://david.ncifcrf.gov/) and 
+[*Pathway Studio*](https://www.pathwaystudio.com).
+
+Although gene set enrichment methods have been primarily developed and
+applied on transcriptomic data, they have recently been modified,
+extended and applied also in other fields of genomic and biomedical
+research. This includes novel approaches for functional enrichment
+analysis of proteomic and metabolomic data as well as genomic regions
+and disease phenotypes [@Lavallee16; @Chagoyen16; @McLean10;@Ried12].
+
+## Gene sets, pathways & regulatory networks
+
+*Gene sets* are simple lists of usually functionally related genes
+without further specification of relationships between genes.
+
+*Pathways* can be interpreted as specific gene sets, typically
+representing a group of genes that work together in a biological
+process. Pathways are commonly divided in metabolic and signaling
+pathways. Metabolic pathways such as glycolysis represent biochemical
+substrate conversions by specific enzymes. Signaling pathways such as
+the MAPK signaling pathway describe signal transduction cascades from
+receptor proteins to transcription factors, resulting in activation or
+inhibition of specific target genes.
+
+*Gene regulatory networks* describe the interplay and effects of
+regulatory factors (such as transcription factors and microRNAs) on the
+expression of their target genes.
+
+## Resources
+
+*GO* (<http://www.geneontology.org>) and *KEGG* (<http://www.genome.jp/kegg>)
```

Comment:
Make GO and KEGG links rather than displaying the URLs.

Reply:
Done